buildenv/jenkins/common/variables-functions

```diff
@@ -47,6 +47,37 @@ def parse_variables_file(){
     VARIABLES = readYaml file: "${VARIABLE_FILE}"
 }
 
+/*
+* Fetch the user provided variable file.
+*/
+def get_variables_file() {
+    if (params.VARIABLE_FILE && params.VENDOR_REPO && params.VENDOR_BRANCH) {
+        stage('Get Variables File') {
+            timestamps {
+                if (params.VENDOR_CREDENTIALS_ID) {
+                    sshagent(credentials:["${params.VENDOR_CREDENTIALS_ID}"]) {
+                        checkout_file()
+                    }
+                } else {
+                    checkout_file()
+                }
+             }
+        }
+    }
+
+    return params.VARIABLE_FILE
+}
+
+/*
+* Check out the user variable file.
+*/
+def checkout_file() {
+    sh "git config remote.vendor.url ${params.VENDOR_REPO}"
+    sh "git config remote.vendor.fetch +refs/heads/*:refs/remotes/vendor/*"
```

Comment:
Is this not the default refspec? You could remove this line or you could change it to only fetch the `VENDOR_BRANCH`
i.e. ```+refs/heads/${params.VENDOR_BRANCH}:refs/remotes/vendor/${params.VENDOR_BRANCH}```

Reply:
It cannot be removed as it is required by the git fetch. It can be restricted to the branch only.

Reply:
Updated refs to fetch branch.

Reply:
Right, tested by hand, when you add a remote using `git config` and then `fetch` you only get `FETCH_HEAD`